badger_db.go

```diff
@@ -28,7 +28,14 @@ func NewBadgerDB(dbName, dir string) (*BadgerDB, error) {
 	if err := os.MkdirAll(path, 0755); err != nil {
 		return nil, err
 	}
+
+	valueThreshold := int64(5 << 20) // 5MB
```

Comment:
I'm not sure where these values came from.  

Are these numbers chosen to address some specific concern?  If so: Is there an issue we could link to understand why these particular values are good enough when the defaults aren't?

I'd prefer to be cautious about overriding the defaults, particularly for a library used on potentially a lot of different systems. Increasing size limits and concurrency can have implications for containerized deployments that have to tune their RAM budget, descriptor limits, etc.

Re: DetectConflicts: Disabling the conflict checking _might_ be safe in some cases, but it does affect the semantics of the implementation. It'd be good to have some tests to exercise concurrent Get/Set/Delete workloads to have confidence this doesn't allow inconsistent views.

Reply:
These... are bad. I'll update the branch, with finalized code :)

Reply:
The value threshold comes from this:

https://github.com/tharsis/evmos/pull/542

I also want to throw a pretty huge red flag on badger:

https://github.com/dgraph-io/badger/pull/1754

If that PR is merged, then we can use badger, and fi not, pls see 542 on evmos-- large genesis.json files will fail.

Reply:
FWIW, Tendermint since v0.35 (specifically after https://github.com/tendermint/tendermint/pull/6554) does not store the genesis document in the database anyway (regardless which backend is used).

I expect applications will have different storage profiles than the core consensus node, and that's fine. One of the goals of decoupling the consensus storage from application storage is to allow those two profiles to be different.